packages/composites/circuit-breaker/src/adapter.ts

```diff
@@ -0,0 +1,65 @@
+import { AdapterError, Requester, util, Validator, Logger } from '@chainlink/ea-bootstrap'
+import { AdapterRequest, ExecuteWithConfig, Config, ExecuteFactory } from '@chainlink/types'
+import { makeConfig } from './config'
+import { AxiosResponse } from 'axios'
+
+const customParams = {
+  primarySource: true,
+  secondarySource: true,
+}
+
+export interface ResponseSchema {
+  result: any
+}
+
+export const execute: ExecuteWithConfig<Config> = async (input, _, config) => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.jobRunID
+  const primarySource = validator.validated.data.primarySource
+  const secondarySource = validator.validated.data.secondarySource
+  const sources = secondarySource ? [primarySource, secondarySource] : [primarySource]
+  const urls = sources.map((source) => util.getRequiredURL(source.toUpperCase()))
+  const response = await getResults(jobRunID, sources, urls, input, config)
+  response.data.result = Requester.validateResultNumber(response.data, ['data', 'result'])
+  return Requester.success(jobRunID, response)
```

Comment:
Can we just return it directly?

```suggestion
  return await getResults(jobRunID, sources, urls, input, config)
```

Reply:
done